solr/core/src/java/org/apache/solr/request/macro/MacroExpander.java

```diff
@@ -136,11 +136,12 @@ private String _expand(String val) {
       }
       else if (idx < 0) {
         if (sb == null) return val;
-        sb.append(val.substring(start));
+        sb.append(val, start, val.length());
         return sb.toString();
       }
 
       // found unescaped "${"
+      int matchedStart = idx;
       idx += macroStart.length();
 
       int rbrace = val.indexOf('}', idx);
```

Comment:
```suggestion
      int rbrace = val.indexOf('}', matchedStart + macroStart.length());
```

similar to line 165 below and then the `idx += macroStart.length();` assignment would not be needed since there would be no use of `idx` between that incrementing assigment and the `idx = start = rbrace + 1;` resetting assignment below.